gnucash/gnome/dialog-price-edit-db.cpp

```diff
@@ -532,9 +532,23 @@ gnc_prices_dialog_add_clicked (GtkWidget *widget, gpointer data)
     {
         if (!gnc_list_length_cmp (comm_list, 1)) // make sure it is only one parent
         {
-            price = gnc_price_create (pdb_dialog->book);
             auto comm = static_cast<gnc_commodity *> (comm_list->data);
-            gnc_price_set_commodity (price, comm);
+            auto latest_price = gnc_pricedb_lookup_latest_any_currency (pdb_dialog->price_db, comm);
+
+            if (latest_price)
+            {
+                price = static_cast<GNCPrice *> (latest_price->data);
+                gnc_price_ref (price);
```

Comment:
This will edit the existing price instead of creating a new one; you want `gnc_price_clone` instead. It's null-checked so it's safe to say
```
price = gnc_price_clone(GNC_PRICE(latest_price->data));
```
It's created with a ref count of 1 so there's no need to ref it. Notice the cast macro: GncPrices are GObjects so you should always use the cast macro that checks its GType instead of a static_cast or C-cast.

Reply:
See gnc_price_edit_dialog on L636:
The price is copied there, only a reference is given. Or am I overseeing something?

```c++
    case GNC_PRICE_NEW:
        if (price)
        {
            price = gnc_price_clone(price, pedit_dialog->book);

            gnc_price_set_source (price, PRICE_SOURCE_EDIT_DLG);
            gnc_price_set_time64 (price, gnc_time (NULL));
            gnc_price_set_value (price, gnc_numeric_zero ());
        }
```

Reply:
I changed the casts to the GLib-Style-Casts as requested.

Reply:
> See gnc_price_edit_dialog on L636

OK. The memory management is a bit tortured but that's not your doing.